### PR TITLE
Add extension to `convert` `ITensorMPS.MPS` to `Tenet.MPS` and viceversa

### LINKED ISSUE #247: convert to/from ITensors MPS
It would be good to be able to convert between the two. An ITensorMPS.MPS is basically a list of ITensors plus a couple integers (llim, rlim) that localize the orthogonality center, but those should be added automatically when you create one. 

The simplest way is probably to convert the tensors of Tenet's MPS into ITensors, and feed the list of ITensors into the ITensorMPS.MPS() constructor. The only thing I don't know how to do in Tenet,as always, is how to map the indices into the tensors.  Here's a (non-working) attempt

```
 psi_tensors = [rand(2,10), rand(2,10,12), rand(2,12,4), rand(2,4)]
 psi_tenet = MPS(psi_tensors)

itensors = [] 
# this assumes some ordering of the Tenet tensor indices. one should also treat first and last separately but you get the idea
for (j,t) in enumerate(tensors(psi_tenet))
   it = ITensor(t, Index(dim("physical index of t"),"Site,n=$j"),Index(dim("left link index of t"),"Link,l=$j"),Index(dim("right link index of t"),"Link,l=$(j+1")
push!(itensors, it)
end 
psi_itensors = ITensorMPS.MPS(itensors)
```

the other way around is probably simpler (for me at least), here's one way to do it 

```
psiten = ITensorMPS.random_mps(siteinds(4, 10), linkdims=7) 

# bring itensors to (o,l,r) and convert them

# first and last always annoying
first_ten = array(psiten[1], siteinds(psiten)[1], linkinds(psiten)[1])
last_ten = array(psiten[end], siteinds(psiten)[end], linkinds(psiten)[end])

ttensors = Array{}[array(psiten[jj], siteinds(psiten)[jj], linkinds(psiten)[jj-1],linkinds(psiten)[jj]) for jj in eachindex(psiten)[2:end-1]]
pushfirst!(ttensors, first_ten)
push!(ttensors,last_ten)

Tenet.MPS(ttensors)
```

### PR DESCRIPTION
**Pull request by jofrevalles**
### Summary
This PR resolves #247 by adding `TenetITensorMPSExt` extension, which adds two `convert` functions to convert `ITensorMPS.MPS` to `Tenet.MPS` and viceversa. Additionally, we also added some tests to cover for this function. @starsfordummies

#### TODO

- [x] Clean some code.

### Example
```julia
julia> using Tenet

julia> using ITensors; using ITensorMPS

julia> tenet_mps = rand(MPS; n=5, maxdim=30)
MPS (inputs=0, outputs=5)

julia> itensor_mps = convert(ITensorMPS.MPS, tenet_mps) # Tenet.MPS to ITensorMPS conversion
ITensors.ITensorMPS.MPS
[1] ((dim=2|id=565|"Site,n=1"), (dim=2|id=954|"Link,l=1"))
[2] ((dim=2|id=557|"Site,n=2"), (dim=2|id=954|"Link,l=1"), (dim=4|id=965|"Link,l=2"))
[3] ((dim=2|id=556|"Site,n=3"), (dim=4|id=965|"Link,l=2"), (dim=4|id=651|"Link,l=3"))
[4] ((dim=2|id=195|"Site,n=4"), (dim=4|id=651|"Link,l=3"), (dim=2|id=558|"Link,l=4"))
[5] ((dim=2|id=263|"Site,n=5"), (dim=2|id=558|"Link,l=4"))

julia> @test isapprox(parent(Tenet.contract(tenet_mps)), Array(ITensorMPS.contract(itensor_mps).tensor))
Test Passed

julia> itensor_mps = ITensorMPS.random_mps(siteinds(4, 5); linkdims=7)
ITensors.ITensorMPS.MPS
[1] ((dim=4|id=51|"Site,n=1"), (dim=7|id=149|"Link,l=1"))
[2] ((dim=7|id=149|"Link,l=1"), (dim=4|id=113|"Site,n=2"), (dim=7|id=746|"Link,l=2"))
[3] ((dim=7|id=746|"Link,l=2"), (dim=4|id=227|"Site,n=3"), (dim=7|id=555|"Link,l=3"))
[4] ((dim=7|id=555|"Link,l=3"), (dim=4|id=400|"Site,n=4"), (dim=4|id=984|"Link,l=4"))
[5] ((dim=4|id=984|"Link,l=4"), (dim=4|id=459|"Site,n=5"))

julia> tenet_mps = convert(MPS, itensor_mps) # ITensorMPS to Tenet.MPS conversion
MPS (inputs=0, outputs=5)

julia> form(tenet_mps)
MixedCanonical(1)

julia> contracted = permutedims(Tenet.contract(tenet_mps), [inds(tenet_mps; at=Site(i)) for i in 1:length(tensors(tenet_mps))])
4×4×4×4×4 Tensor{Float64, 5, Array{Float64, 5}}: ...

julia> @test isapprox(parent(contracted), Array(ITensorMPS.contract(itensor_mps).tensor))
Test Passed
```

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -66,6 +68,7 @@ EinExprs = "0.5, 0.6"
 FiniteDifferences = "0.12"
 GraphMakie = "0.4,0.5"
 Graphs = "1.7"
+ITensorMPS = "0.2.6"
```
Comment (mofeing) [inline Project.toml:71]: mmm if we set the patch version, will we have problems when v0.2.7 lands?

like, will it force it to use v0.2.6 or will it be compatible (but not included) up to v0.3.0?

in principle we should only mark major and minor versions, not patch version (unless a patch is completely required for it to work)

**Review by mofeing — COMMENTED** [review]
super cool!

File: Project.toml (hunk 1 above)
Comment (mofeing) [reply Project.toml:71]: I think we should try to just set the major.minor version. If you need a higher patch version, you should just call `Pkg.update()` then.

**Review by starsfordummies — APPROVED** [review]
great, thanks!

**Review by mofeing — APPROVED** [review]

**Comment by mofeing** [comment]
The tests failing are my fault. If they pass locally, you can merge.

**Pull request merged**